app/models/user.rb

```diff
@@ -32,17 +33,31 @@ def remember
     update_attributes remember_digest: User.digest(remember_token)
   end
 
-  def authenticated? remember_token
-    return false if remember_digest.nil?
-    BCrypt::Password.new(remember_digest).is_password?(remember_token)
+  def authenticated? attribute, token
+    digest = send "#{attribute}_digest"
```

Comment:
giải thích giúp anh cách hoạt động của method này nhé, từng dòng một

Reply:
#{attribute} được coi như để giữ chỗ.
Tùy vào attribute: có thể là thuộc tính hoặc method.
Ví dụ trong tường hợp trên nó sẽ trả về user.activation_digest
Trong db có : remember_digest, activation_digest, password_digest, reset_digest

Reply:
meta programming, not siêu dữ liệu